test/localPRG_test.cpp

```diff
@@ -272,6 +275,17 @@ TEST(LocalPRGTest, nodes_along_path) {
     EXPECT_EQ(j, l3.nodes_along_path(p3)[1]->id);
 }
 
+TEST(LocalPRGTest, nodes_along_path_without_memoization) {
+    assert(LocalPRG::path_memoization_in_nodes_along_path_method == false);
+    nodes_along_path_core_test();
+}
+
+TEST(LocalPRGTest, nodes_along_path_with_memoization) {
+    LocalPRG::path_memoization_in_nodes_along_path_method=true;
+    nodes_along_path_core_test();
+    LocalPRG::path_memoization_in_nodes_along_path_method=false;
```

Comment:
This test doesn't seem to test anything? There is no assertion of anything being equal or true etc.

Reply:
All tests are done on the `nodes_along_path_core_test` function (https://github.com/leoisl/pandora/blob/improving_compare/test/localPRG_test.cpp#L163-L276), which is called in this test. It tests a bunch of things (i.e. it was the previous unit test). On `TEST(LocalPRGTest, nodes_along_path_without_memoization)`, we just make sure memoization is turned off and call `nodes_along_path_core_test()` (i.e. we test `LocalPRG::nodes_along_path()` without memoization - exactly what was done before). This test tests `LocalPRG::nodes_along_path()` with memoization, to make sure memoization does not introduce any bugs.